### PR TITLE
Fix pylint and pickling issue

### PR DESCRIPTION
**Pull request by ahartikainen**
## Checklist

- [ ] Follows [official](https://github.com/arviz-devs/arviz/blob/main/CONTRIBUTING.md#pull-request-checklist) PR format
- [ ] Includes a sample plot to visually illustrate the changes (only for plot-related functions)
- [ ] New features are properly documented (with an example if appropriate)?
- [ ] Includes new or updated tests to cover the new feature
- [ ] Code style  correct (follows pylint and black guidelines)
- [ ] Changes are listed in [changelog](https://github.com/arviz-devs/arviz/blob/main/CHANGELOG.md#v0xx-unreleased)

### CODE REVIEW

**Comment by canyon289** [comment]
This has a new issue now right? PyMC is causing some caching problem?

**Comment by ahartikainen** [comment]
Apparently. Usually these fix themselves in a day or two.

**Comment by canyon289** [comment]
I ran tests locally in multiple envs and they passed so rerunning here

**Comment by canyon289** [comment]
Noting here as well. PyMC changed to cloud pickle which points to why pickle is failing. Should we make the switch as well?

https://github.com/pymc-devs/pymc3/blob/main/pymc3/tests/test_model.py#L410

File: arviz/data/io_pyro.py [hunk 1]
```diff
@@ -108,6 +108,7 @@ def arbitrary_element(dct):
 
         observations = {}
         if self.model is not None:
+            # pylint: disable=not-callable
```
Comment (MarcoGorelli) [inline arviz/data/io_pyro.py:111]: this'll disable the `not-callable` check everywhere in the file below this line - I presume the intention was just to disable it for `pyro.poutine.trace`? If so, perhaps
```suggestion
            # pylint: disable=not-callable
            trace = pyro.poutine.trace(self.model).get_trace(*self._args, **self._kwargs)
            # pylint: enable=not-callable
```
or
```suggestion
            trace = pyro.poutine.trace(self.model).get_trace(*self._args, **self._kwargs)  # pylint: disable=not-callable
```
would be better?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

File: arviz/data/io_pyro.py (hunk 1 above)
Comment (canyon289) [reply arviz/data/io_pyro.py:111]: Ohh thanks, Ill change this

**Comment by canyon289** [comment]
Alright I dont have enough time to figure out mypy but propose we merge this so we can unblock all other PRs. Thoughts?

File: .azure-pipelines/azure-pipelines-external.yml [hunk 2]
```diff
@@ -87,13 +87,12 @@ jobs:
...
       if [ "$(pymc3.version)" = "github" ]; then
           # Pip installation is failing for some reason. This is the same thing
-          git clone https://github.com/pymc-devs/pymc3
-          cd pymc3
-          git checkout v4
-          cd ..
-          pip install $PWD/pymc3
-          # python -m pip --no-cache-dir --log log.txt install git+https://github.com/pymc-devs/pymc3
-          # cat log.txt
+          # git clone https://github.com/pymc-devs/pymc3 cd pymc3
```
Comment (canyon289) [inline .azure-pipelines/azure-pipelines-external.yml:90]: We can delete the "manual" install method now.

File: .azure-pipelines/azure-pipelines-external.yml [hunk 3]
```diff
@@ -132,25 +131,26 @@ jobs:
...
-    condition: failed()
-    displayName: 'update mypy.ini with typing_copilot'
-
-  - task: PublishBuildArtifacts@1
-    condition: failed()
-    inputs:
-      pathtoPublish: 'mypy.ini'
-      artifactName: 'mypy.ini'
-    displayName: 'Upload updated mypy.ini'
+# Disable temporarily until we find a fix
```
Comment (canyon289) [inline .azure-pipelines/azure-pipelines-external.yml:134]: Im not sure how to get the mypy stuff to work with cloud pickle. Was getting a ci failure about not finding the types for cloudpickle library but dont know how to fix it

Comment (MarcoGorelli) [reply .azure-pipelines/azure-pipelines-external.yml:134]: you probably just need to tell mypy to ignore missing imports from it

Comment (MarcoGorelli) [reply .azure-pipelines/azure-pipelines-external.yml:134]: basically, what you do here
```
[mypy-theano.*]
ignore_missing_imports = True
```
, do that for cloudpickle too

Comment (canyon289) [reply .azure-pipelines/azure-pipelines-external.yml:134]: Where do I add that code? In the mypy.ini file?

Comment (canyon289) [reply .azure-pipelines/azure-pipelines-external.yml]: Seems like I gotta add it there! Trying it out. Thank you marco

**Pull request merged**